FirebaseStorage/Tests/Unit/FIRStorageListTests.m

```diff
@@ -79,6 +79,29 @@ - (void)testValidatesInput {
   [FIRStorageTestHelpers waitForExpectation:self];
 }
 
+- (void)testValidatesListAllCallback {
+  XCTestExpectation *expectation =
+      [self expectationWithDescription:@"testValidatesListAllCallback"];
+  expectation.expectedFulfillmentCount = 1;
+
+  FIRStoragePath *path = [FIRStorageTestHelpers objectPath];
+  FIRStorageReference *ref = [[FIRStorageReference alloc] initWithStorage:self.storage path:path];
+
+  FIRStorageVoidListError errorBlock = ^(FIRStorageListResult *result, NSError *error) {
+    XCTAssertNil(result);
+    XCTAssertNotNil(error);
+
+    XCTAssertEqualObjects(error.domain, @"FIRStorageErrorDomain");
+    XCTAssertEqual(error.code, FIRStorageErrorCodeUnknown);
```

Comment:
It's not immediately obvious to me why this test is throwing an error. Should we make it more explicit and follow the pattern in `testListWithErrorResponse`?

Reply:
I followed the pattern of the test above it which is sufficient to execute the error code paths. Unlike the API tested in `testListWithErrorResponse`, `listAll` manages the service implicitly, so more mocking would be needed for a success case or different error.